doc/userguide/devguide/extending/app-layer/overview.rst

```diff
@@ -0,0 +1,98 @@
+**************************
+Application Layer Overview
+**************************
+
+.. contents:: Table of Contents
+
+This section aims to give an overview of what is needed to add
+an application-layer protocol to Suricata.
+
+After a generic first step of collecting data about this application-layer protocol,
+especially pcaps for testing, we can dive into the Suricata specifics.
+
+An application-layer protocol has three logic components in Suricata :
+- parser
+- logger
+- detection engine
+
+Both detection engine and logger will depend on the processing done by the parser.
```

Comment:
Suggestion to add:
```suggestion
Both detection engine and logger will depend on the processing done by the parser.
As such, first the parser should be added, then the logger, and only then the detection module.
```

Reply:
I do not agree. I think it should be done altogether... (promoting log-keywords parity)

Reply:
Fair, but even for testing purposes, shouldn't first the logger be added, and then the detecting module? Even if part of the same work?

Reply:
People will not edit the 2 files detect and logger at once. But I still think the result should be one commit.
If we document that logging should be done first, detection later, we will get PRs with only logging...